spec/classes/init_spec.rb

```diff
@@ -105,6 +119,33 @@
           is_expected.to contain_file('/etc/rhsm/rhsm.conf').with_content(%r{^no_proxy = proxy.local$})
         end
       end
+
+      context 'with specific plugin options' do
+        let :params do
+          {
+            rh_password: 'password',
+            rh_user: 'username',
+            plugin_settings: { 'main' => { 'enabled' => 0, 'disable_system_repos' => 1 } }
+          }
+        end
+
+        if facts[:os]['family'] == 'RedHat' && facts[:os]['release']['major'] == 7
```

Comment:
`facts[:os]['release']['major']` is a string, you should either convert it to integer with `to_i` or use `versioncmp` to compare the to versions.

Also RedHat 6 falls in the else branch which according to the puppet code is wrong, that's why redhat 6 are failing.